test/JDBC/expected/restricting_permission_on_role.out

```diff
@@ -0,0 +1,167 @@
+grant sysadmin to restricting_permission_on_role_u1;
+go
+~~ERROR (Code: 0)~~
+
+~~ERROR (Message: ERROR: must have admin option on role "sysadmin"
+    Server SQLState: 42501)~~
+
+
+-- Altering a role by an underprivileged login should be restricted
+alter user restricting_permission_on_role_u1 with password '123'
+go
+~~ERROR (Code: 0)~~
+
+~~ERROR (Message: ERROR: permission denied
+    Server SQLState: 42501)~~
+
+
+-- Dropping a role by an underprivileged login should be restricted
+drop user restricting_permission_on_role_u1;
+go
+~~ERROR (Code: 0)~~
+
+~~ERROR (Message: ERROR: permission denied to drop role
+    Server SQLState: 42501)~~
+
+
+-- psql user=restricting_permission_on_role_u1 password=123
+-- Granting sysadmin membership by an underprivileged login should be restricted
+grant sysadmin to restricting_permission_on_role_l1;
+go
+~~ERROR (Code: 0)~~
+
+~~ERROR (Message: ERROR: must have admin option on role "sysadmin"
+    Server SQLState: 42501)~~
+
+
+-- Creating user by an underprivileged login should be restricted
+create user restricting_permission_on_role_u2;
+go
+~~ERROR (Code: 0)~~
+
+~~ERROR (Message: ERROR: permission denied to create role
+    Server SQLState: 42501)~~
+
+
+-- Granting sysadmin membership by an underprivileged login should be restricted
+grant sysadmin to restricting_permission_on_role_u1;
+go
+~~ERROR (Code: 0)~~
+
+~~ERROR (Message: ERROR: must have admin option on role "sysadmin"
+    Server SQLState: 42501)~~
+
+
+-- Altering a role by an underprivileged login should be restricted
+alter user restricting_permission_on_role_l1 with password '123'
+go
+~~ERROR (Code: 0)~~
+
+~~ERROR (Message: ERROR: permission denied
+    Server SQLState: 42501)~~
+
+
+-- Dropping a role by an underprivileged login should be restricted
+drop user restricting_permission_on_role_u1;
+go
+~~ERROR (Code: 0)~~
+
+~~ERROR (Message: ERROR: permission denied to drop role
+    Server SQLState: 42501)~~
+
+
+-- tsql
+alter server role sysadmin add member restricting_permission_on_role_l1;
+go
+
+-- psql user=restricting_permission_on_role_l1 password=123
+-- user has sysadmin membership via TDS Port, create user is allowed
+create user restricting_permission_on_role_u2 with password '123';
+go
+
+-- user has sysadmin membership, alter user is allowed
+alter user restricting_permission_on_role_u2 with password '1234'
+go
+
+-- user has sysadmin membership, drop user is allowed
+drop user restricting_permission_on_role_u2;
+go
+
+-- tsql
+alter server role sysadmin drop member restricting_permission_on_role_l1;
+go
+
+-- psql
+-- Grant sysadmin privilege to underprivileged T-SQL user
+grant sysadmin to restricting_permission_on_role_l1;
+go
+
+-- Grant sysadmin privilege to underprivileged PG user
+grant sysadmin to restricting_permission_on_role_u1;
+go
+
+-- psql user=restricting_permission_on_role_l1 password=123
+-- user has sysadmin membership via PG port, create user is not allowed
```

Comment:
At here and all other places, Let's use the terms tsql login or psql user to avoid confusion about which one we talking.